compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -3849,22 +3849,9 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
         end implicitArgs
 
         val args = implicitArgs(wtp.paramInfos, 0, pt)
-
-        def propagatedFailure(args: List[Tree]): Type = args match {
-          case arg :: args1 =>
-            arg.tpe match {
-              case ambi: AmbiguousImplicits =>
-                propagatedFailure(args1) match {
-                  case NoType | (_: AmbiguousImplicits) => ambi
-                  case failed => failed
-                }
-              case failed: SearchFailureType => failed
-              case _ => propagatedFailure(args1)
-            }
-          case Nil => NoType
-        }
-
-        val propFail = propagatedFailure(args)
+        val firstAmbiguous = args.tpes.find(_.isInstanceOf[AmbiguousImplicits])
+        def firstError = args.tpes.find(_.isError)
+        val propFail = firstAmbiguous.orElse(firstError).getOrElse(NoType)
```

Comment:
So the previous logic prefer non-ambiguity errors over ambiguity errors, and this PR changes it to prefer ambiguity errors over non-ambiguity errors, but since ambiguity is no longer special after https://dotty.epfl.ch/docs/reference/changed-features/implicit-resolution.html, perhaps we could just report the first error found? /cc @odersky